runtime/src/chpl-cache.c

```diff
@@ -666,27 +634,77 @@ struct rdcache_s {
   chpl_comm_nb_handle_t *pending;
   cache_seqn_t *pending_sequence_numbers;
 
-  // space for mid-level entries
-  int max_top_entries;
-  struct cache_entry_base_s* free_top_nodes_head; // a linked list.
-
-  // The entry into the 'pointer tree' hashtable structure.
-  struct top_entry_s* top_index_list[TOP_SIZE];
+  // padding to get table 64-byte aligned
```

Comment:
Did you consider (and reject) `__attribute__ ((aligned (64)))` here?  It seems to be fairly widely acceptable despite originating with gcc.  We use it in runtime/include/atomics/intrinsics/chpl-atomics.h, for example, as well as the ugni comm layer.  It would actually align things, as opposed to simply padding them into the next cache line for example.

Reply:
It would be fine with me to use it if we're using it elsewhere. I wouldn't want to be the first place in the runtime to use something like that...

But these are allocated all at once and then individual pointers pulled out of the buffer. I think the only way we'd pad to the next cache line is when adding fields to this struct. But we could check the alignment somewhere with an assertion, alternatively.

Reply:
To actually answer your question - no I hadn't considered it.

Reply:
I think that __attribute__ would be the more principled way to achieve the goal, but it's your call.  I'll approve it as it stands.

Reply:
I will do that in a separate PR just in case there is an issue with it.

Reply:
(also we do already check the alignment with an assertion, FWIW)

Reply:
PR #16734 will do it.